File: articles/resilience-adaptation-and-long-horizon-decisions/julia/resilience_stock_model.jl
# resilience_stock_model.jl

function resilience_update(current, recovery, investment, degradation, shock)
    return max(0.0, current + recovery + investment - degradation - shock)
end

println("Next resilience stock = ", round(resilience_update(35.0, 3.0, 2.0, 1.0, 1.6), digits=6))

File: articles/resilience-adaptation-and-long-horizon-decisions/python/resilience_stock_model.py
#!/usr/bin/env python3
"""Resilience stock helper."""

from __future__ import annotations


def resilience_update(current: float, recovery: float, investment: float, degradation: float, shock: float) -> float:
    return max(0.0, current + recovery + investment - degradation - shock)


if __name__ == "__main__":
    print(round(resilience_update(35.0, 3.0, 2.0, 1.0, 1.6), 6))
